examples/text_labels_and_annotations/rainbow_text.py

```diff
@@ -51,7 +51,9 @@ def rainbow_text(x, y, strings, colors, orientation='horizontal',
     if ax is None:
         ax = plt.gca()
     t = ax.transData
-    canvas = ax.figure.canvas
+    fig = ax.figure
+    canvas = fig.canvas
+    dpi = fig.dpi
```

Comment:
I think this is close but still fragile.  Instead I think you should use fig.scale_dpi_trans.transform_bbox (ahem or maybe it's inverse) to get the extent in inches.  Yes under the hood it's the same thing but I think it doesn't depend on fig.dpi properly tracking the renderers dpi.

Reply:
I changed the solution to use ```fig.dpi_scale_trans.inverted().transform_bbox```. Feel free to let me know if I misunderstood what you were saying. Thank you for the advice!